src/geouned/GEOUNED/Conversion/CellDefinition.py

```diff
@@ -598,7 +598,7 @@ def GenTorusAnnexVSurface(face,Vparams,forceCylinder=False):
        Apex   = face.Surface.Center + za * axis
        semiAngle = abs(math.atan(d1/(z1-za)))
 
-       ConeAxis = axis if za < 0 else -axis
+       ConeAxis = axis if (face.Surface.Center.dot(axis) + z1 - Apex.dot(axis)) > 0.0 else -axis
```

Comment:
You are right the current test is wrong although it worked so far.
I would prefer you to rewrite the instruction like 
```ConeAxis = axis if z1-za > 0.0 else -axis```